WordPress/wporg-developer
# Add Block Editor Handbook page template

This is a first-pass at adding in the `single.html` from Documentation, for use on the handbook pages. Right now this only applies to the Block Editor Handbook, I don't have any page content in the other handbooks locally to test. This is a start for #161, but does not fix it because we still need the left column handbook nav. 

This also serves as a demo for https://github.com/WordPress/wporg-mu-plugins/pull/321.

**Screenshots**

| Desktop | Mobile |
|-------|------|
| ![fullpage](https://user-images.githubusercontent.com/541093/213035450-89008887-c807-4282-9139-4db3c7eb614f.png) | ![mobile](https://user-images.githubusercontent.com/541093/213035736-b5786cf4-210c-4ec8-8df5-6fc4ba8bfc51.png) |

**To test**

1. View a block editor handbook page, like `http://localhost:8888/block-editor/getting-started/create-block/block-code/`

To remove the old Table of Contents I added the same filter that [I added to the support-helphub plugin](https://meta.trac.wordpress.org/changeset/12327/) which uses this same code (not literally, unfortunately). Replace `load_filters` in `source/wp-content/plugins/handbook/inc/table-of-contents.php` with this:

```php
public function load_filters() {
	$page_supports_toc = is_singular( $this->post_types ) && ! is_embed() && ! is_front_page();
	/**
	 * Filter whether the table of contents should be injected into the page.
	 *
	 * @param bool $page_supports_toc True if the current page supports a table of contents.
	 */
	$should_add_toc = apply_filters( 'wporg_handbook_toc_should_add_toc', $page_supports_toc );
	if ( $should_add_toc ) {
		add_filter( 'the_content', array( $this, 'add_toc' ) );
	}
}
```
